runtime/commonMain/src/kotlinx/serialization/internal/Primitives.kt

```diff
@@ -5,19 +5,21 @@
 package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
-import kotlin.reflect.KClass
+import kotlin.reflect.*
 
 @InternalSerializationApi
-public sealed class PrimitiveDescriptor(override val name: String, override val kind: PrimitiveKind) :
-    SerialDescriptor {
-    private fun error(): Nothing = throw IllegalStateException("Primitives does not have elements")
-
+public sealed class PrimitiveDescriptor( // TODO internal constructor
```

Comment:
it is sealed anyway, do we really need internal constructor?

Reply:
true